core/market/decentralized/src/matcher/store.rs

```diff
@@ -58,6 +58,37 @@ impl SubscriptionStore {
             })
     }
 
+    pub async fn get_offers(&self, id: Option<Identity>) -> Result<Vec<ClientOffer>, OfferError> {
+        Ok(self
+            .db
+            .as_dao::<OfferDao>()
+            .get_offers(match id {
+                Some(ident) => Some(ident.identity),
+                _ => None,
+            })
+            .await
+            .map_err(|e| OfferError::GetMany(e))?
+            .into_iter()
+            .filter_map(|o| match o.into_client_offer() {
+                Err(e) => {
+                    log::error!("Skipping Offer because of: {}", e);
+                    None
+                }
+                Ok(o) => Some(o),
+            })
+            .collect())
+    }
+
+    pub async fn get_offers_before(&self, demand: &Demand) -> Result<Vec<Offer>, OfferError> {
+        let now = Utc::now().naive_utc();
+        Ok(self
+            .db
+            .as_dao::<OfferDao>()
+            .get_offers_before(demand.insertion_ts.unwrap(), now)
+            .await
+            .map_err(|e| OfferError::GetMany(e))?)
+    }
+
```

Comment:
Good point.

The intent was to show the purpose or the context in which this is usable. `insertion_ts` is designed for that single purpose to be compared against other Demand[s], not between Offers themselves. 

But now I see I can go even further and enclose getting single Demand and all Offers before it within single offer store fn. Since it is the only usecase for it and will illustrate the intent clearly. Furthermore, it will ensure that `unwrap` is safe.

Reply:
but for the time being, I'm simplifying the API as you've suggested